src/modules/esl-share/core/esl-share-trigger.ts

```diff
@@ -1,37 +1,157 @@
-import {ESLTrigger} from '../../esl-trigger/core';
+import {ESLBaseElement} from '../../esl-base-element/core';
+import {attr, boolAttr, jsonAttr, listen} from '../../esl-utils/decorators';
+import {ENTER, SPACE} from '../../esl-utils/dom/keys';
+import {DeviceDetector} from '../../esl-utils/environment/device-detector';
+import {ESLMediaQuery} from '../../esl-media-query/core';
+import {ESLTraversingQuery} from '../../esl-traversing-query/core';
+import {ESLSharePopup} from './esl-share-popup';
 
-import type {ESLToggleableActionParams} from '../../esl-toggleable/core/esl-toggleable';
+import type {ESLSharePopupActionParams} from './esl-share-popup';
 
 /**
  * ESLShareTrigger component
  * @author Dmytro Shovchko
  *
- * ESLShareTrigger is an extension of {@link ESLTrigger} that
- * - controls an internal popup of {@link ESLShare} module in `popup` rendering mode.
- * - forwards the sharing attributes from the parent share {@link ESLShare} component to its associated instance of {@link ESLPopup}
+ * ESLShareTrigger is an component that allows triggering {@link ESLSharePopup} instance state changes.
  */
-export class ESLShareTrigger extends ESLTrigger {
+export class ESLShareTrigger extends ESLBaseElement {
```

Comment:
Is it so different from the ESLTrigger?

Reply:
Probably not, not much.
Is it worth a try? Or shall we discuss it first?